### PR TITLE
CASM-3981: Add S3 artifacts and Loftsman manifests to the Product Catalog schema

### PR DESCRIPTION
**Pull request by lathanm**
- Added S3 artifacts and Loftsman manifests to the Product Catalog schema

- Added functions to retrieve Helm charts, S3 artifacts, and Loftsman manifests from cray-product-catalog
This part of code was extracted from the [PR276](https://github.com/Cray-HPE/cray-product-catalog/pull/276) as this change is required for product-deletion-utility code base

## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [issue id](issue link)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Target branch correct
- [ ] Testing is appropriate and complete, if applicable
- [ ] Is a new version being released? Update https://github.com/Cray-HPE/cray-product-catalog/wiki/CSM-Compatibility-Matrix
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by arnab-muk-hpe — APPROVED** [review]
Approved from my side. Not able to see split related logics as decided.

**Review by paulose-kuriakose — APPROVED** [review]

**Review by krabhi-hpe — APPROVED** [review]
LGTM.

**Review by atifsyedali — APPROVED** [review]

**Review by paulose-kuriakose — APPROVED** [review]

**Review by krabhi-hpe — APPROVED** [review]

**Review by arnab-muk-hpe — DISMISSED** [review]

**Comment by lathanm** [comment]
> Approving based on there will be an upversion (you can upversion minor because there are additions of API).

Updated version from 1.8.12 to 1.9.0

**Review by krabhi-hpe — APPROVED** [review]

**Review by arnab-muk-hpe — DISMISSED** [review]

**Review by mharding-hpe — COMMENTED** [review]
Rename .helm to match other properties

File: tests/mocks.py [hunk 1]
```diff
@@ -74,14 +74,17 @@
 
 # Two versions of a product named COS where:
 # - The two versions have one docker image name and version in common
-# - The first version has no repositories, configuration, images, or recipes
-# - The second version has repositories, configuration, images, and recipes
+# - The first version has docker images and manifests but not helm charts, repositories, configuration, images, or recipes
+# - The second version has docker images, helm charts, repositories, configuration, images, and recipes, but not manifests
```
Comment (mharding-hpe) [inline tests/mocks.py:78]:
```suggestion
# - The first version has docker images and manifests but not helm charts, repositories, configuration,
    images, or recipes
# - The second version has docker images, helm charts, repositories, configuration, images, and recipes,
    but not manifests
```

File: tests/mocks.py [hunk 2]
```diff
@@ -74,14 +74,19 @@
 
 # Two versions of a product named COS where:
 # - The two versions have one docker image name and version in common
-# - The first version has no repositories, configuration, images, or recipes
-# - The second version has repositories, configuration, images, and recipes
+# - The first version has docker images and manifests but not helm charts, repositories, configuration,
+    images, or recipes
+# - The second version has docker images, helm charts, repositories, configuration, images, and recipes,
+    but not manifests
```
Comment (mharding-hpe) [inline tests/mocks.py:80]:
```suggestion
#   images, or recipes
# - The second version has docker images, helm charts, repositories, configuration, images, and recipes,
#   but not manifests
```

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**